tools/codegen/codespec/api_to_provider_spec_mapper_test.go

```diff
@@ -19,16 +26,78 @@ func TestConvertToProviderSpec(t *testing.T) {
+					Schema: &codespec.Schema{
+						Description: conversion.StringPtr(testResourceDesc),
+						Attributes: codespec.Attributes{
+							codespec.Attribute{
+								Name:                     "bool_default_attr",
+								ComputedOptionalRequired: codespec.ComputedOptional,
+								Bool:                     &codespec.BoolAttribute{Default: conversion.Pointer(false)},
+							},
+							codespec.Attribute{
+								Name:                     "count",
+								ComputedOptionalRequired: codespec.Optional,
+								Int64:                    &codespec.Int64Attribute{},
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+							codespec.Attribute{
+								Name:                     "create_date",
+								String:                   &codespec.StringAttribute{},
+								ComputedOptionalRequired: codespec.Computed,
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+							codespec.Attribute{
+								Name:                     "group_id",
+								ComputedOptionalRequired: codespec.Required,
+								String:                   &codespec.StringAttribute{},
+								Description:              conversion.StringPtr(testPathParamDesc),
+							},
+							codespec.Attribute{
+								Name:                     "num_double_default_attr",
+								Float64:                  &codespec.Float64Attribute{Default: conversion.Pointer(2.0)},
+								ComputedOptionalRequired: codespec.ComputedOptional,
+							},
+							codespec.Attribute{
+								Name:                     "str_computed_attr",
+								ComputedOptionalRequired: codespec.Computed,
+								String:                   &codespec.StringAttribute{},
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+							codespec.Attribute{
+								Name:                     "str_req_attr1",
+								ComputedOptionalRequired: codespec.Required,
+								String:                   &codespec.StringAttribute{},
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+							codespec.Attribute{
+								Name:                     "str_req_attr2",
+								ComputedOptionalRequired: codespec.Required,
+								String:                   &codespec.StringAttribute{},
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+							codespec.Attribute{
+								Name:                     "str_req_attr3",
+								String:                   &codespec.StringAttribute{},
+								ComputedOptionalRequired: codespec.Required,
+								Description:              conversion.StringPtr(testFieldDesc),
+							},
+						},
+					},
+					Name: "test_resource",
+				},
+				},
+			},
 		},
 	}
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
-			result := codespec.ToProviderSpecModel(tc.inputOpenAPISpecPath, tc.inputConfigPath, &tc.inputResourceName)
+			result, _ := codespec.ToProviderSpecModel(tc.inputOpenAPISpecPath, tc.inputConfigPath, tc.inputResourceName)
```

Comment:
instead of ignoring error, maybe we can assert it is nil?

Reply:
updated to use require.NoError(t, err)